From 1074d797ae30ebb151a55492f8f5e0fe93b53716 Mon Sep 17 00:00:00 2001
From: Mark Charlebois <charlebm@gmail.com>
Date: Tue, 20 Nov 2012 21:21:12 +0100
Subject: [PATCH] Kbuild: LLVMLinux: Disable the use of the Integrated
 Assembler when compiling with Clang

Globally disable the use of the Clang Integrated Assembler (IA) unless
IA=1 is set. If IA=1 is not set, use gas instead. IA doesn't support
the gas assembly extensions used in the kernel code.

Author: Mark Charlebois <charlebm@gmail.com>
Signed-off-by: Mark Charlebois <charlebm@gmail.com>
Signed-off-by: Behan Webster <behanw@converseincode.com>
---
 Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 2893d7f..89def73 100644
--- a/Makefile
+++ b/Makefile
@@ -386,7 +386,10 @@
 ifneq ($(GCC_TOOLCHAIN),)
 CLANG_GCC_TC	:= -gcc-toolchain $(GCC_TOOLCHAIN)
 endif
-CLANG_FLAGS	:= $(CLANG_TARGET) $(CLANG_GCC_TC)
+ifneq ($(IA),1)
+CLANG_IA_FLAG	= -no-integrated-as
+endif
+CLANG_FLAGS	:= $(CLANG_TARGET) $(CLANG_GCC_TC) $(CLANG_IA_FLAG)
 endif
 
 
-- 
1.9.1

